root/static/scripts/genre/components/GenreEditForm.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import MB from '../../common/MB';
+import EnterEdit from '../../../../components/EnterEdit';
+import EnterEditNote from '../../../../components/EnterEditNote';
+import FormRowTextLong from '../../../../components/FormRowTextLong';
+import type {GenreFormT} from '../../../../genre/types';
+import {createExternalLinksEditor} from '../../edit/externalLinks';
+import {exportTypeInfo} from '../../relationship-editor/common/viewModel';
+import {prepareSubmission} from '../../relationship-editor/generic';
+
+type Props = {
+  +$c: CatalystContextT,
+  +attrInfo: LinkAttrTypeOptionsT,
+  +form: GenreFormT,
+  +sourceEntity: GenreT | {entityType: 'genre'},
+  +typeInfo: LinkTypeOptionsT,
+};
+
+const GenreEditForm = ({
+  $c,
+  attrInfo,
+  form,
+  sourceEntity,
+  typeInfo,
+}: Props): React.Element<'form'> => {
+  const externalLinksEditorContainerRef = React.useRef(null);
+
+  const handleSubmit = () => {
+    console.log('preparing submission');
+    prepareSubmission('edit-genre');
+  };
+
+  React.useEffect(() => {
+    exportTypeInfo(typeInfo, attrInfo);
+    invariant(externalLinksEditorContainerRef.current != null);
+    // $FlowIgnore
+    MB.sourceExternalLinksEditor = createExternalLinksEditor({
```

Comment:
This effect should also have a cleanup handler where we unmount the external links editor root it creates. 
 (`createExternalLinksEditor` will have to return the root too.)

You can already see that an error is logged due to this effect running twice in development mode.

To register a cleanup function, just return a function from the effect. I guess something like this is the idea:

```JS
const {root, externalLinksEditorRef} = createExternalLinksEditor(...);

MB.sourceExternalLinksEditor = externalLinksEditorRef;

return () => {
  root.unmount();
};

Reply:
I tried that, but it only replaced the warning with a different one:

`Warning: Attempted to synchronously unmount a root while React was already rendering. React cannot finish unmounting the root until the current render has completed, which may lead to a race condition.`